python/usb_console/radio_session.py

```diff
@@ -59,27 +72,78 @@ def check_for_downlink(self):
         Read device output for debug messages and state variable updates. Record debug messages
         to the logging file, and update the console's record of the state.
         '''
+        
+        #connect to PAN email account  
+        mail = imaplib.IMAP4_SSL("imap.gmail.com", 993)
+        mail.login(self.username, self.password)
+        mail.select('Inbox')
 
         while self.running_logger:
-            # TODO implement
-            pass
+            #.search() searches from mail. Data gives id's of all emails.
+            type, data = mail.search(None, '(FROM "sbdservice@sbd.iridium.com")', '(UNSEEN)')
+            mail_ids = data[0]
+            id_list = mail_ids.split()
+
+            for num in id_list:
+                #.fetch() fetches the mail for given id where 'RFC822' is an Internet 
+                # Message Access Protocol.
+                typ, data = mail.fetch(num,'(RFC822)')
+                
+                #go through each component of data and print sender and subject line
+                for response_part in data:
```

Comment:
It might be better to split off the actual email processing into a different function in the class--there's a lot of nested for-loops here.